services/streammanager/wunderkind/wunderkindmanager_test.go

```diff
@@ -0,0 +1,147 @@
+	"github.com/rudderlabs/rudder-server/services/streammanager/common"
+)
+
+var (
+	sampleMessage    = "sample payload"
+	sampleFunction   = "sampleLambdaFunction"
+	sampleExternalID = "sampleExternalID"
+	sampleIAMRoleARN = "sampleRoleArn"
+	invocationType   = "RequestResponse"
+)
+
+func TestNewProducer(t *testing.T) {
+	t.Setenv("WUNDERKIND_REGION", "us-east-1")
+	t.Setenv("WUNDERKIND_IAM_ROLE_ARN", sampleIAMRoleARN)
+	t.Setenv("WUNDERKIND_EXTERNAL_ID", sampleExternalID)
+	t.Setenv("WUNDERKIND_LAMBDA", sampleFunction)
+
+	destinationConfig := map[string]interface{}{}
+	destination := backendconfig.DestinationT{
+		Config:      destinationConfig,
+		WorkspaceID: "sampleWorkspaceID",
+	}
+	timeOut := 10 * time.Second
+	producer, err := NewProducer(&destination, common.Opts{Timeout: timeOut})
+	assert.Nil(t, err)
+	assert.NotNil(t, producer)
+	assert.NotNil(t, producer.client)
+
+	// Invalid Region
+	t.Setenv("WUNDERKIND_EXTERNAL_ID", "")
+	destinationConfig = map[string]interface{}{}
+	destination.Config = destinationConfig
+	timeOut = 10 * time.Second
+	producer, err = NewProducer(&destination, common.Opts{Timeout: timeOut})
+	assert.Nil(t, producer)
+	assert.Equal(t, "externalID is required for IAM role", err.Error())
+}
+
+func TestProduceWithInvalidClient(t *testing.T) {
+	producer := &WunderkindProducer{}
+	sampleEventJson := []byte("{}")
+	statusCode, statusMsg, respMsg := producer.Produce(sampleEventJson, map[string]string{})
+	assert.Equal(t, 400, statusCode)
+	assert.Equal(t, "Failure", statusMsg)
+	assert.Equal(t, "[Wunderkind] error :: Could not create client", respMsg)
+}
+
+func TestProduceWithInvalidData(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	mockClient := mock_lambda.NewMockLambdaClient(ctrl)
+	producer := &WunderkindProducer{client: mockClient}
+	mockLogger := mock_logger.NewMockLogger(ctrl)
+	pkgLogger = mockLogger
+
+	// Invalid input
+	sampleEventJson := []byte("invalid json")
+	statusCode, statusMsg, respMsg := producer.Produce(sampleEventJson, map[string]string{})
+	assert.Equal(t, 400, statusCode)
+	assert.Equal(t, "Failure", statusMsg)
+	assert.Contains(t, respMsg, "[Wunderkind] error while unmarshalling jsonData ")
+
+	// Empty payload
+	sampleEventJson, _ = json.Marshal(map[string]interface{}{
+		"payload": "",
+	})
+	statusCode, statusMsg, respMsg = producer.Produce(sampleEventJson, map[string]string{})
+	assert.Equal(t, 400, statusCode)
+	assert.Equal(t, "Failure", statusMsg)
+	assert.Contains(t, respMsg, "[Wunderkind] error :: Invalid payload")
+}
+
+func TestProduceWithServiceResponse(t *testing.T) {
+	t.Setenv("WUNDERKIND_REGION", "us-east-1")
+	t.Setenv("WUNDERKIND_IAM_ROLE_ARN", sampleIAMRoleARN)
+	t.Setenv("WUNDERKIND_EXTERNAL_ID", sampleExternalID)
+	t.Setenv("WUNDERKIND_LAMBDA", sampleFunction)
+
+	ctrl := gomock.NewController(t)
+	mockClient := mock_lambda.NewMockLambdaClient(ctrl)
+	producer := &WunderkindProducer{client: mockClient}
+	mockLogger := mock_logger.NewMockLogger(ctrl)
```

Comment:
Use mem logger instead, much more maintainable.

Reply:
Are you referring about this?
https://pkg.go.dev/go.chromium.org/luci/common/logging/memlogger

Reply:
@lvrach We don't have mem logger implementation. But in this particular case, we can simply asssert for returned variables, why do we need to verify if we are logging or not?